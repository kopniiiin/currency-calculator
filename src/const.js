export const Currency = {
  EUR: `EUR`,
  USD: `USD`,
  JPY: `JPY`
};

export const currenciesToTranslations = {
  [Currency.EUR]: `ÐµÐ²Ñ€Ð¾`,
  [Currency.USD]: `Ð´Ð¾Ð»Ð»Ð°Ñ€`,
  [Currency.JPY]: `Ð¸ÐµÐ½Ð°`
};

export const currenciesToEmoji = {
  [Currency.EUR]: `ðŸ’¶`,
  [Currency.USD]: `ðŸ’µ`,
  [Currency.JPY]: `ðŸ’´`
};

export const RATE_PRECISION = 2;
