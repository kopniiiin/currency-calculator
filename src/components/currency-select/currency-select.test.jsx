import React from "react";
import {shallow} from "enzyme";

import {Currency} from "../../const.js";

import CurrencySelect, {Mode} from "./currency-select.jsx";

describe(`snapshot test: CurrencySelect`, () => {
  it(`should render in original mode`, () => expect(
    shallow(
      <CurrencySelect
        mode={Mode.ORIGINAL}
        activeCurrency={Currency.EUR}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());

  it(`should render in final mode`, () => expect(
    shallow(
      <CurrencySelect
        mode={Mode.FINAL}
        activeCurrency={Currency.EUR}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());
});
