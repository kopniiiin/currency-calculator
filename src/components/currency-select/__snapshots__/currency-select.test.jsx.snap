// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: CurrencySelect should render in final mode 1`] = `
<div
  aria-label="итоговая валюта"
  className=" currency-select"
  role="group"
>
  <CurrencyRadioInput
    checked={true}
    className=""
    currency="EUR"
    key="EUR"
    label="евро"
    name="final-currency"
    onChange={[Function]}
    value="EUR"
  />
  <CurrencyRadioInput
    checked={false}
    className=""
    currency="USD"
    key="USD"
    label="доллар"
    name="final-currency"
    onChange={[Function]}
    value="USD"
  />
  <CurrencyRadioInput
    checked={false}
    className=""
    currency="JPY"
    key="JPY"
    label="иена"
    name="final-currency"
    onChange={[Function]}
    value="JPY"
  />
</div>
`;

exports[`snapshot test: CurrencySelect should render in original mode 1`] = `
<div
  aria-label="исходная валюта"
  className=" currency-select"
  role="group"
>
  <CurrencyRadioInput
    checked={true}
    className=""
    currency="EUR"
    key="EUR"
    label="евро"
    name="original-currency"
    onChange={[Function]}
    value="EUR"
  />
  <CurrencyRadioInput
    checked={false}
    className=""
    currency="USD"
    key="USD"
    label="доллар"
    name="original-currency"
    onChange={[Function]}
    value="USD"
  />
  <CurrencyRadioInput
    checked={false}
    className=""
    currency="JPY"
    key="JPY"
    label="иена"
    name="original-currency"
    onChange={[Function]}
    value="JPY"
  />
</div>
`;
